### DISCUSSION TITLE
Create a Guest Order?

### DISCUSSION THREAD

**Discussion by osseonews** [swellstores/community#42] [Help] [answered]
Your Create Account API, https://developers.swell.is/backend-api/orders/create-an-order, says that an Account ID is required for an order. If so, how do you create guest orders on the frontend? We have situations where users don't want to have an account with Swell and just want to checkout as guests. Is this not possible? Do we need to create an account on the backend for every order, even if its a guest account? I see that the cart itself has a "guest" boolean field, so presumably people can checkout as guests. if so, why is Account ID required for an order and what do we use there for a guest order?

**Comment by ericingram** [comment, marked as answer]
From a data model point of view, an account is considered a guest account if it doesn't have a password. The `guest` field is set automatically if the account does not have a password.

**Reply by osseonews** [reply]
thanks. wasn't aware that a password is optional.

**Comment by sebastian-meckovski** [comment]
I was able to successfully submit orders both as a guest and as a logged-in user using the following approach:

```javascript
import swell from "swell-js";
...
onClick={async () => {
  console.log("Order placed");

  if (isLoggedIn) {
    await swell.cart.submitOrder();
  } else {
    await swell.account.create({
      email: "some-example@email.com",
    });
    await swell.cart.submitOrder();
  }
}}
```
In the case of a guest checkout, I'm creating a user account without a password if the user is not logged in.

**The problem:**
This flow only works once. If the same user tries to perform a guest checkout again using the same email address, Swell throws an error because an account with that email already exists.

**My question:**
Is there a way to allow guest checkout even if the email is already associated with an existing account?
Or is the intended behavior that returning users must log in if they previously checked out as a guest?

Any insight or recommended practices would be appreciated!

**Reply by sebastian-meckovski** [reply]
found a solution

```javascript
  await swell.cart.update({
    account: {
      email: "some-example@email.com",
    },
```
